src/sql/workbench/contrib/query/browser/keyboardQueryActions.ts

```diff
@@ -210,18 +213,24 @@ export class RunCurrentQueryWithActualPlanKeyboardAction extends Action {
 	public static ID = 'runCurrentQueryWithActualPlanKeyboardAction';
 	public static LABEL = nls.localize('runCurrentQueryWithActualPlanKeyboardAction', "Run Current Query with Actual Plan");
 
+	private executionPlanTelemetry: ExecutionPlanTelemetryHelper;
+
 	constructor(
 		id: string,
 		label: string,
-		@IEditorService private _editorService: IEditorService
+		@IEditorService private _editorService: IEditorService,
+		@IInstantiationService instantiationService: IInstantiationService
 	) {
 		super(id, label);
 		this.enabled = true;
+		this.executionPlanTelemetry = instantiationService.createInstance(ExecutionPlanTelemetryHelper);
 	}
 
 	public override run(): Promise<void> {
 		const editor = this._editorService.activeEditorPane;
 		if (editor instanceof QueryEditor) {
+			let planOptions = { displayActualQueryPlan: true } as ExecutionPlanOptions;
+			this.executionPlanTelemetry.addTelemetry(TelemetryKeys.TelemetryAction.RunQuery, editor.input.uri, planOptions);
```

Comment:
You shouldn't be sending a new event for this - you should add the necessary props to the RunQuery event I linked to in queryManagement